### PR TITLE
TST: Add tests string series min max

### LINKED ISSUE #31746: New string data type aggregations (min, max, sum) work for DataFrames but not Series
```python
>>> s = pd.Series(['a', 'b'], dtype='string')
>>> s.max()
TypeError: Cannot perform reduction 'max' with string dtype
>>> df = s.to_frame()
>>> df.max()
0    b
dtype: object
```
#### Problem description

I assume this isn't supposed to work on DataFrames if it doesn't work on strings. min and sum do the same thing.

#### Expected Output

Consistent behavior for both dataframes and series

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.8.1.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 19.2.0
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 0+untagged.1.gce8af21.dirty
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 45.1.0.post20200127
Cython           : 0.29.14
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.5.0
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.10.3
IPython          : 7.11.1
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.5.0
matplotlib       : 3.1.1
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : None
pyxlsb           : None
s3fs             : None
scipy            : 1.3.1
sqlalchemy       : 1.3.13
tables           : None
tabulate         : 0.8.3
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by NumberPiOso**
Add tests for max and min aggregation at string series

In line with #31746

- [X] closes #31746
- [X] tests added / passed
- [X] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them

### CODE REVIEW

File: pandas/tests/series/test_reductions.py [hunk 1]
```diff
@@ -9,6 +9,13 @@
 import pandas._testing as tm
 
 
+def test_reductions_series_strings():
+    # GH#31746
+    ser = Series(["a", "b"], dtype="string")
+    assert ser.max() == "b"
```
Comment (mroeschke) [inline pandas/tests/series/test_reductions.py:15]: Could you use `pytest.mark.parameterize` over `min/max` and the expected result?

Comment (NumberPiOso) [reply pandas/tests/series/test_reductions.py:15]: Of course!

File: pandas/tests/series/test_reductions.py [hunk 2]
```diff
@@ -9,6 +9,17 @@
 import pandas._testing as tm
 
 
+@pytest.mark.parametrize("operation", ["min", "max"])
+def test_reductions_series_strings(operation):
+    # GH#31746
+    list_str = ["a", "b"]
+    ser = Series(list_str, dtype="string")
+    res_operation_serie = getattr(ser, operation)()
+    func_builtin_python = __builtins__[operation]
```
Comment (mroeschke) [inline pandas/tests/series/test_reductions.py:18]: Hmm instead of this could you hardcore the expected (a or b) with the parameterize as well?

**Review by mroeschke — APPROVED** [review]

**Comment by jreback** [comment]
thanks @NumberPiOso

**Pull request merged**